### PR TITLE
Add incremental Slack ingestion using LastIngest timestamp

### PR DESCRIPTION
**Pull request by scalabilitysolved**
## Summary

Slack ingestion currently scans the full 90-day lookback window on every run, even if the source was ingested an hour ago. The checksum deduplication avoids re-processing unchanged documents, but all the Slack API calls to fetch messages are still made every time. For high-volume channels this is wasteful and slow.

This PR makes Slack ingestion incremental by plumbing the source's `LastIngest` timestamp through to the connector. On subsequent ingestions, the Slack connector now queries only from `LastIngest - 1 day` instead of the full 90-day window, dramatically reducing API calls. The 1-day overlap buffer guards against messages posted near the boundary, daily digest documents that straddle it, and clock skew.

The `LastIngest` field is added to `ScanOptions` at the public connector interface level, so any connector *can* use it, but only Slack does — filesystem, git, and Confluence connectors simply ignore it. The pipeline and remote ingestion paths both look up the source record from the store before scanning. `--force` bypasses the optimization so forced re-ingestion always scans the full window.

## Test plan

New tests cover the Slack connector using `LastIngest` to narrow the query window, falling back to the 90-day default on first scan, the pipeline passing `LastIngest` from the store, the `--force` flag bypassing it, and the edge case where a source exists but has a zero `LastIngest`. All existing tests continue to pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/connector/connector.go [hunk 1]
```diff
@@ -13,6 +14,12 @@ type ScanOptions struct {
 	// Known maps paths to their checksums for incremental ingestion.
 	// The connector may skip files whose checksum hasn't changed.
 	Known map[string]string
+
+	// LastIngest is the timestamp of the last successful ingestion for this source.
+	// Connectors that support time-based filtering (e.g., Slack) can use this
+	// to narrow the scan window instead of always fetching the full lookback period.
+	// Zero value means no previous ingestion (first scan).
+	LastIngest time.Time
```
[the remark this answers is not in the document]
Comment (alecgard) [reply pkg/connector/connector.go]: Claude is dead https://status.claude.com/
